airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py

```diff
@@ -369,31 +369,33 @@ def wait_for_job(self, job, stream_state: Mapping[str, Any] = None) -> AdReportR
 
             if job["async_status"] == "Job Completed":
                 return job
-            elif job["async_status"] in ["Job Failed", "Job Skipped"]:
+            else:
                 time_range = (job["date_start"], job["date_stop"])
-                if self.times_job_restarted.get(time_range, 0) < 6:
+                self.times_job_restarted[time_range] = self.times_job_restarted.get(time_range, 0)
+                if self.times_job_restarted[time_range] < 6:
                     params = deep_merge(
                         {"time_range": {"since": job["date_start"], "until": job["date_stop"]}},
                         self.request_params(stream_state=stream_state),
                     )
                     restart_job = self._create_insights_job(params)
                     self.running_jobs.append(restart_job)
                     self.times_job_restarted[time_range] += 1
-                elif job["async_status"] == "Job Failed":
-                    raise JobTimeoutException(f"AdReportRun {job} failed after {runtime.in_seconds()} seconds.")
-                elif job["async_status"] == "Job Skipped":
-                    raise JobTimeoutException(f"AdReportRun {job} skipped after {runtime.in_seconds()} seconds.")
-
-            if runtime > self.MAX_WAIT_TO_START and job_progress_pct == 0:
-                raise JobTimeoutException(
-                    f"AdReportRun {job} did not start after {runtime.in_seconds()} seconds."
-                    f" This is an intermittent error which may be fixed by retrying the job. Aborting."
-                )
-            elif runtime > self.MAX_WAIT_TO_FINISH:
-                raise JobTimeoutException(
-                    f"AdReportRun {job} did not finish after {runtime.in_seconds()} seconds."
-                    f" This is an intermittent error which may be fixed by retrying the job. Aborting."
-                )
+                else:
+                    if job["async_status"] == "Job Failed":
+                        raise JobTimeoutException(f"AdReportRun {job} failed after {runtime.in_seconds()} seconds.")
```

Comment:
Please do not use JobTimeoutException here, this error is not related to timeout and very confusing.